Userland/Libraries/LibCrypto/Curves/SECPxxxr1.h

```diff
@@ -0,0 +1,700 @@
+
+        StorageType r = 0u;
+        StorageType s = 0u;
+        for (size_t i = 0; i < (KEY_BIT_SIZE / 32); i++) {
+            StorageType rr = r_bigint.words()[i];
+            StorageType ss = s_bigint.words()[i];
+            r |= (rr << (i * 32));
+            s |= (ss << (i * 32));
+        }
+
+        // z is the hash
+        StorageType z = import_big_endian(hash.slice(0, KEY_BYTE_SIZE));
+
+        StorageType r_mo = to_montgomery_order(r);
+        StorageType s_mo = to_montgomery_order(s);
+        StorageType z_mo = to_montgomery_order(z);
+
+        StorageType s_inv = modular_inverse_order(s_mo);
+
+        StorageType u1 = modular_multiply_order(z_mo, s_inv);
+        StorageType u2 = modular_multiply_order(r_mo, s_inv);
+
+        u1 = from_montgomery_order(u1);
+        u2 = from_montgomery_order(u2);
+
+        auto u1_buf = TRY(ByteBuffer::create_uninitialized(KEY_BYTE_SIZE));
+        export_big_endian(u1, u1_buf.bytes());
+        auto u2_buf = TRY(ByteBuffer::create_uninitialized(KEY_BYTE_SIZE));
+        export_big_endian(u2, u2_buf.bytes());
+
+        auto p1 = TRY(generate_public_key(u1_buf));
+        auto p2 = TRY(compute_coordinate(u2_buf, pubkey));
+
+        JacobianPoint point1 {
+            import_big_endian(TRY(p1.slice(1, KEY_BYTE_SIZE))),
+            import_big_endian(TRY(p1.slice(1 + KEY_BYTE_SIZE, KEY_BYTE_SIZE))),
+            1u,
+        };
+
+        // Convert the input point into Montgomery form
+        point1.x = to_montgomery(point1.x);
+        point1.y = to_montgomery(point1.y);
+        point1.z = to_montgomery(point1.z);
+
+        VERIFY(is_point_on_curve(point1));
+
+        JacobianPoint point2 {
+            import_big_endian(TRY(p2.slice(1, KEY_BYTE_SIZE))),
+            import_big_endian(TRY(p2.slice(1 + KEY_BYTE_SIZE, KEY_BYTE_SIZE))),
+            1u,
+        };
+
+        // Convert the input point into Montgomery form
+        point2.x = to_montgomery(point2.x);
+        point2.y = to_montgomery(point2.y);
+        point2.z = to_montgomery(point2.z);
+
+        VERIFY(is_point_on_curve(point2));
+
+        JacobianPoint result;
+        point_add(result, point1, point2);
+
+        // Convert from Jacobian coordinates back to Affine coordinates
+        convert_jacobian_to_affine(result);
+
+        // Make sure the resulting point is on the curve
+        VERIFY(is_point_on_curve(result));
+
+        // Convert the result back from Montgomery form
+        result.x = from_montgomery(result.x);
+        result.y = from_montgomery(result.y);
+        // Final modular reduction on the coordinates
+        result.x = modular_reduce(result.x);
+        result.y = modular_reduce(result.y);
+
+        return r.is_equal_to_constant_time(result.x);
+    }
+
+private:
+    constexpr StorageType import_big_endian(ReadonlyBytes bytes)
+    {
+        VERIFY(bytes.size() == KEY_BYTE_SIZE);
+
+        StorageType result = 0u;
+        for (size_t i = 0; i < bytes.size(); i++) {
```

Comment:
You can `memcpy` (`TypedTransfer`) bytes and use `{Little,Big}Endian<UFixedBigInt>`.

Reply:
Doing the following seems to work:
```cpp
BigEndian<StorageType> result;
memcpy(&result, bytes.data(), KEY_BYTE_SIZE);
return result;
```
However, how would you use `TypedTransfer` in this case?

Reply:
Another option I see would be to use a `MemoryStream` like this:
```cpp
AK::FixedMemoryStream stream { bytes };
auto result = stream.read_value<BigEndian<StorageType>>();
return result.release_value();
```
This might also be nice in other places where we are reading multiple things sequentially from a buffer of bytes.

Reply:
I guess you can leave `memcpy` unless maintainers have strong opinion on this. With `TypedTransfer`, it will look like
```cpp
StorageType result;
TypedTransfer::copy(result.bytes().data(), bytes.data(), KEY_BYTE_SIZE);
return BigEndian<StorageType>(result);
```

Reply:
Ignore my previous code, it will swap bytes two times (and will do nothing)